consensus_pledge_model/logic.py

```diff
@@ -289,37 +289,37 @@ def s_sectors_rewards(_1,
     & reward_schedule_init = {0: 20, 1: 30, 2: 40, 3: 50}
     then
     reward_schedule_final = {0: 20 + 5, 1: 30 + 5, 2: 40 + 5, 3: 50}
-    
+
     """
-    #retrieve total rewards
-    total_reward = Reward("block_reward")
-    linear_duration = ConsensusPledgeParams("linear_duration")
+    # retrieve total rewards
+    total_reward = state["reward"]
+    linear_duration = params["linear_duration"]
     current_sector_list = state["aggregate_sectors"]
-    total_qa = AggregateSectorList.power_qa()
-    sector_qa = AggregateSector("qa_power")
-    reward_schedule = AggregateSector{reward_schedule}
-    
+    total_qa = state["power_qa"]
+    sector_qa = state["aggregate_sectors"]
+    reward_schedule = state["aggregate_sectors"]
+    immediate_release = params["immediate_release_fraction"]
+    days_passed = state['days_passed']
+
     for agg_sector in current_sector_list:
-        #get share of total reward
-        share_qa = sector_qa / total_qa 
-        share_reward = share_qa * total_reward
+        # get share of total reward
+        share_qa = sector_qa / total_qa
+        available_reward = total_reward * (1.0 - immediate_release)
+        share_reward = share_qa * available_reward
         daily_reward = share_reward / linear_duration
-        #create new reward schedule dict to be merged
-        today_reward_schedule = {k:daily_reward for k in range(linear_duration)}
-        #pop 0 day from old reward schedule
-        reward_schedule.pop(0)
-        #create new list of dict items to iterate over
-        newlist = list(reward_schedule.items())
-        #create new dict from list with 1 subtracted from keys to reorder to day0 
-        reward_schedule = {k-1 : v for k,v in newlist}
-
-        #create new dict, and adds the values from shifted reward schedule and 
-        #newly created reward schedule
-        updated_reward_schedule = {x: reward_schedule.get(x, 0) + today_reward_schedule.get(x, 0)
-        for x in set(reward_schedule).union(today_reward_schedule)}
+        # create new reward schedule dict to be merged
+        today_reward_schedule = {
+            k + days_passed: daily_reward for k in range(linear_duration)}
+        # new method of transforming the dict
+        new_reward_schedule = {
+            k: v for k, v in reward_schedule.items() if k > days_passed}
+
+        # create new dict, and adds the values from shifted reward schedule and
+        # newly created reward schedule
+        updated_reward_schedule = {x: new_reward_schedule.get(x, 0.0) + today_reward_schedule.get(x, 0.0)
+                                   for x in set(new_reward_schedule).union(today_reward_schedule)}
```

Comment:
Style suggestion: set an variable called `reward_days = set(new_reward_schedule) | set(today_reward_schedule)`, and do the dict comprehension on top of it, eg. `{... for unlock_day in reward_days}`